### PR TITLE
Remove ALL protected virtual DataPortal_XYZ methods attempt #2

### LINKED ISSUE #2164: Remove DataPortal_XYZ methods from base classes
With the new data portal operation attributes (such as `Fetch` and `Insert`) the old DataPortal_XYZ methods are no longer required, and just complicate code in and out of the framework.

Remove all `protected virtual` DataPortal_XYZ operation methods.

### PR DESCRIPTION
**Pull request by adrianwright109**
Fixes #2164

### CODE REVIEW

File: Source/Csla/DynamicBindingListBase.cs [hunk 1]
```diff
@@ -432,23 +432,6 @@ protected override void OnDeserialized()
...
-    /// object based on data in the database.
-    /// </summary>
-    /// <param name="criteria">An object containing criteria values to identify the object.</param>
-    [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Naming", "CA1707:IdentifiersShouldNotContainUnderscores", MessageId = "Member")]
-    protected virtual void DataPortal_Fetch(object criteria)
-    {
-      throw new NotSupportedException(Properties.Resources.FetchNotSupportedException);
-    }
-
     private void DataPortal_Update()
```
Comment (rockfordlhotka) [inline Source/Csla/DynamicBindingListBase.cs:435]: Shouldn't this method go away also?

File: Source/Csla/DynamicListBase.cs [hunk 2]
```diff
@@ -497,23 +497,6 @@ protected override void OnDeserialized()
...
-    /// object based on data in the database.
-    /// </summary>
-    /// <param name="criteria">An object containing criteria values to identify the object.</param>
-    [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Naming", "CA1707:IdentifiersShouldNotContainUnderscores", MessageId = "Member")]
-    protected virtual void DataPortal_Fetch(object criteria)
-    {
-      throw new NotSupportedException(Properties.Resources.FetchNotSupportedException);
-    }
-
     private void DataPortal_Update()
```
Comment (rockfordlhotka) [inline Source/Csla/DynamicListBase.cs:500]: Shouldn't this method go away also?

File: Source/Csla/NameValueListBase.cs [hunk 3]
```diff
@@ -264,23 +264,6 @@ protected virtual object GetClone()
...
-    /// object based on data in the database.
-    /// </summary>
-    /// <param name="criteria">An object containing criteria values to identify the object.</param>
-    [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Naming", "CA1707:IdentifiersShouldNotContainUnderscores", MessageId = "Member")]
-    protected virtual void DataPortal_Fetch(object criteria)
-    {
-      throw new NotSupportedException(Resources.FetchNotSupportedException);
-    }
-
     private void DataPortal_Update()
```
Comment (rockfordlhotka) [inline Source/Csla/NameValueListBase.cs:267]: Shouldn't this method go away also?

File: Source/Csla/ReadOnlyBindingListBase.cs [hunk 4]
```diff
@@ -96,23 +96,6 @@ public T Clone()
...
-    /// object based on data in the database.
-    /// </summary>
-    /// <param name="criteria">An object containing criteria values to identify the object.</param>
-    [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Naming", "CA1707:IdentifiersShouldNotContainUnderscores", MessageId = "Member")]
-    protected virtual void DataPortal_Fetch(object criteria)
-    {
-      throw new NotSupportedException(Resources.FetchNotSupportedException);
-    }
-
     private void DataPortal_Update()
```
Comment (rockfordlhotka) [inline Source/Csla/ReadOnlyBindingListBase.cs:99]: Shouldn't this method go away also?

File: Source/Csla/ReadOnlyListBase.cs [hunk 5]
```diff
@@ -103,23 +103,6 @@ public C[] ToArray()
...
-    /// object based on data in the database.
-    /// </summary>
-    /// <param name="criteria">An object containing criteria values to identify the object.</param>
-    [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Naming", "CA1707:IdentifiersShouldNotContainUnderscores", MessageId = "Member")]
-    protected virtual void DataPortal_Fetch(object criteria)
-    {
-      throw new NotSupportedException(Resources.FetchNotSupportedException);
-    }
-
     private void DataPortal_Update()
```
Comment (rockfordlhotka) [inline Source/Csla/ReadOnlyListBase.cs:106]: Shouldn't this method go away also?

**Review by rockfordlhotka — APPROVED** [review]

**Comment by rockfordlhotka** [comment]
@adrianwright109 this is looking good, nice work!

**Comment by adrianwright109** [comment]
@rockfordlhotka - I believe this is now ready for final review

**Pull request merged**